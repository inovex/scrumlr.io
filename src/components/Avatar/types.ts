export const AVATAR_SKIN_COLORS = ["Tanned", "Yellow", "Pale", "Light", "Brown", "DarkBrown", "Black"] as const;
export type AvatarSkinColor = (typeof AVATAR_SKIN_COLORS)[number];

export const AVATAR_TOP_TYPES = [
  "Eyepatch",
  "Hat",
  "Hijab",
  "LongHairBigHair",
  "LongHairBob",
  "LongHairBun",
  "LongHairCurly",
  "LongHairCurvy",
  "LongHairDreads",
  "LongHairFrida",
  "LongHairFro",
  "LongHairFroBand",
  "LongHairMiaWallace",
  "LongHairNotTooLong",
  "LongHairShavedSides",
  "LongHairStraight",
  "LongHairStraight2",
  "LongHairStraightStrand",
  "NoHair",
  "ShortHairDreads01",
  "ShortHairDreads02",
  "ShortHairFrizzle",
  "ShortHairShaggyMullet",
  "ShortHairShortCurly",
  "ShortHairShortFlat",
  "ShortHairShortRound",
  "ShortHairShortWaved",
  "ShortHairSides",
  "ShortHairTheCaesar",
  "ShortHairTheCaesarSidePart",
  "Turban",
  "WinterHat1",
  "WinterHat2",
  "WinterHat3",
  "WinterHat4",
] as const;
export type AvatarTopType = (typeof AVATAR_TOP_TYPES)[number];

export const AVATAR_CLOTHE_COLORS = [
  "Black",
  "Blue01",
  "Blue02",
  "Blue03",
  "Gray01",
  "Gray02",
  "Heather",
  "PastelBlue",
  "PastelGreen",
  "PastelOrange",
  "PastelRed",
  "PastelYellow",
  "Pink",
  "Red",
  "White",
] as const;
export type AvatarClotheColor = (typeof AVATAR_CLOTHE_COLORS)[number];

export const AVATAR_GRAPHIC_TYPES = ["Bat", "Cumbia", "Deer", "Diamond", "Hola", "Pizza", "Resist", "Selena", "Bear", "SkullOutline", "Skull"] as const;
export type AvatarGraphicType = (typeof AVATAR_GRAPHIC_TYPES)[number];

export const AVATAR_CLOTHE_TYPES = ["BlazerShirt", "BlazerSweater", "CollarSweater", "GraphicShirt", "Hoodie", "Overall", "ShirtCrewNeck", "ShirtScoopNeck", "ShirtVNeck"] as const;
export type AvatarClotheType = (typeof AVATAR_CLOTHE_TYPES)[number];

export const AVATAR_HAIR_COLORS = ["Auburn", "Black", "Blonde", "BlondeGolden", "Brown", "BrownDark", "PastelPink", "Blue", "Platinum", "Red", "SilverGray"] as const;
export type AvatarHairColor = (typeof AVATAR_HAIR_COLORS)[number];

export const AVATAR_FACIAL_HAIR_COLORS = ["Auburn", "Black", "Blonde", "BlondeGolden", "Brown", "BrownDark", "Platinum", "Red"] as const;
export type AvatarFacialHairColor = (typeof AVATAR_FACIAL_HAIR_COLORS)[number];

export const AVATAR_FACIAL_HAIR_TYPES = ["Blank", "BeardMedium", "BeardLight", "BeardMajestic", "MoustacheFancy", "MoustacheMagnum"] as const;
export type AvatarFacialHairType = (typeof AVATAR_FACIAL_HAIR_TYPES)[number];

export const AVATAR_ACCESSORIES_TYPES = ["Blank", "Kurt", "Prescription01", "Prescription02", "Round", "Sunglasses", "Wayfarers"] as const;
export type AvatarAccessoriesType = (typeof AVATAR_ACCESSORIES_TYPES)[number];

export const AVATAR_EYE_TYPES = ["Close", "Cry", "Default", "Dizzy", "EyeRoll", "Happy", "Hearts", "Side", "Squint", "Surprised", "Wink", "WinkWacky"] as const;
export type AvatarEyeType = (typeof AVATAR_EYE_TYPES)[number];

export const AVATAR_EYEBROW_TYPES = [
  "Angry",
  "AngryNatural",
  "Default",
  "DefaultNatural",
  "FlatNatural",
  "FrownNatural",
  "RaisedExcited",
  "RaisedExcitedNatural",
  "SadConcerned",
  "SadConcernedNatural",
  "UnibrowNatural",
  "UpDown",
  "UpDownNatural",
] as const;
export type AvatarEyebrowType = (typeof AVATAR_EYEBROW_TYPES)[number];

export const AVATAR_MOUTH_TYPES = ["Concerned", "Default", "Disbelief", "Eating", "Grimace", "Sad", "ScreamOpen", "Serious", "Smile", "Tongue", "Twinkle", "Vomit"] as const;
export type AvatarMouthType = (typeof AVATAR_MOUTH_TYPES)[number];
