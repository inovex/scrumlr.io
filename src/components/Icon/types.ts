export type IconNames =
  | 'phase1-create'
  | 'phase1-communicate'
  | 'phase1-combine'
  | 'phase2-vote'
  | 'phase3-discuss'
  | 'phase3-define'
  | 'circle-arrow-left'
  | 'circle-arrow-right'
  | 'circle-add'
  | 'circle'
  | 'more'
  | 'plus'
  | 'circle-selection'
  | 'circle-selection-grey'
  | 'logout'
  | 'download'
  | 'edit'
  | 'donate'
  | 'pencil'
  | 'focus'
  | 'close'
  | 'close20'
  | 'close-circle'
  | 'feedback'
  | 'key-up'
  | 'key-down'
  | 'key-esc'
  | 'settings'
  | 'board'
  | 'trash'
  | 'chevron-left'
  | 'chevron-right'
  | 'stack-top'
  | 'stack-mid'
  | 'stack-hover'
  | 'arrow-left'
  | 'arrow-right';
