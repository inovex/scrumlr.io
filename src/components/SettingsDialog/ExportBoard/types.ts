import {Color} from "constants/colors";

export interface BoardType {
  id: string;
  name: string;
  showAuthors: boolean;
  showVoting: string;
  accessPolicy: string;
}

export interface ColumnType {
  id: string;
  name: string;
  color: Color;
  visible: boolean;
}

export interface NoteType {
  id: string;
  author: string;
  text: string;
  position: {
    column: string;
    stack: string | null;
    rank: number;
  };
}

export interface ParticipantType {
  role: string;
  user: {
    id: string;
    name: string;
  };
}

export interface VotingType {
  votes: {
    votes: number;
    votesPerNote: [];
  };
}

export interface BoardDataType {
  board: BoardType;
  columns: ColumnType[];
  notes: NoteType[];
  participants: ParticipantType[];
  votings: VotingType[];
}
