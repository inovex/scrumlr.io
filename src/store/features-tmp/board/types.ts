import {Participant} from "../../features/participants/types";
import {Request} from "../../features/requests/types";
import {Column} from "../../features/columns/types";
import {Note} from "../../features/notes/types";
import {Reaction} from "../../features/reactions/types";
import {Vote} from "../../features/votes/types";
import {Voting} from "../../features/votings/types";

export enum AccessPolicy {
  "PUBLIC" = 0,
  "BY_PASSPHRASE" = 1,
  "BY_INVITE" = 2,
}

export interface Board {
  id: string;

  name?: string;
  accessPolicy: keyof typeof AccessPolicy;
  showAuthors: boolean;
  showNotesOfOtherUsers: boolean;
  showNoteReactions: boolean;
  allowStacking: boolean;
  isLocked: boolean;
  timerStart?: Date;
  timerEnd?: Date;

  sharedNote?: string;
  showVoting?: string;
}

export type BoardActionType = {
  board: Board;
  participants: Participant[];
  requests: Request[];
  columns: Column[];
  notes: Note[];
  reactions: Reaction[];
  votes: Vote[];
  votings: Voting[];
};

export type EditBoardRequest = Partial<Omit<Board, "id">> & {passphrase?: string};

export type BoardStatus = "unknown" | "pending" | "ready" | "rejected" | "accepted" | "passphrase_required" | "incorrect_passphrase" | "too_many_join_requests" | "banned";

export interface BoardState {
  status: BoardStatus;
  data?: Board;
}
