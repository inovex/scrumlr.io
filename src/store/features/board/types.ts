import {Participant} from "../participants";
import {Request} from "../requests";
import {Column} from "../columns";
import {Note} from "../notes";
import {Reaction} from "../reactions";
import {Vote} from "../votes";
import {Voting} from "../votings";

export enum AccessPolicy {
  "PUBLIC" = 0,
  "BY_PASSPHRASE" = 1,
  "BY_INVITE" = 2,
}

export interface Board {
  id: string;

  name?: string;
  accessPolicy: keyof typeof AccessPolicy;
  showAuthors: boolean;
  showNotesOfOtherUsers: boolean;
  showNoteReactions: boolean;
  allowStacking: boolean;
  isLocked: boolean;
  timerStart?: Date;
  timerEnd?: Date;

  sharedNote?: string;
  showVoting?: string;
}

export type BoardActionType = {
  board: Board;
  participants: Participant[];
  requests: Request[];
  columns: Column[];
  notes: Note[];
  reactions: Reaction[];
  votes: Vote[];
  votings: Voting[];
};

export type BoardWithServerTimeOffset = {board: Board; serverTimeOffset: number};

export type EditBoardRequest = Partial<Omit<Board, "id">> & {passphrase?: string};

export type BoardStatus = "unknown" | "pending" | "ready" | "rejected" | "accepted" | "passphrase_required" | "incorrect_passphrase" | "too_many_join_requests" | "banned";

export interface BoardState {
  status: BoardStatus;
  data?: Board;
}
