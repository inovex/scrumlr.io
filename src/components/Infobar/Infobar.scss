@import "constants/style.scss";

.info-bar {
  position: fixed;
  display: flex;
  align-items: center;
  bottom: auto;
  top: calc($column__border-width + $header__height);
  left: 50%;
  transform: translateX(-50%);
  height: $info-bar__height;
  justify-content: center;
  gap: $margin--small;
  z-index: $infobar-z-index;
}

.info-bar__return-to-presented-note-button > svg {
  background-color: $menu-icon-background-color--dark;
  color: $color-white;
}
[theme="dark"] .info-bar__return-to-presented-note-button > svg {
  background-color: $color-white;
  color: $menu-icon-background-color--dark;
}
