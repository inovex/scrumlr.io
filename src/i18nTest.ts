import {join} from "path";
import i18n from "i18next";
import {initReactI18next} from "react-i18next";
import Backend from "i18next-fs-backend";

i18n
  .use(Backend)
  .use(initReactI18next)
  .init({
    backend: {
      loadPath: join(__dirname, "../public/locales/en/translation.json"),
    },
    initImmediate: false,
    lng: "en",
    debug: false,
    react: {
      useSuspense: false,
    },
  });

export default i18n;
