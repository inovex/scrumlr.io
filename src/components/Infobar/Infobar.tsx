import {Timer} from "components/Timer";
import {VoteDisplay} from "components/Votes/VoteDisplay";
import ReactDOM from "react-dom";
import "./Infobar.scss";
import _ from "underscore";
import {useAppSelector} from "store";
import Parse from "parse";

export const InfoBar = () => {
  const state = useAppSelector(
    (applicationState) => ({
      endTime: applicationState.board.data?.timerUTCEndTime,
      activeVoting: applicationState.board.data?.voting === "active",
      possibleVotes: applicationState.voteConfiguration.voteLimit,
      usedVotes: applicationState.votes.filter((vote) => vote.votingIteration === applicationState.voteConfiguration.votingIteration && vote.user === Parse.User.current()?.id)
        .length,
    }),
    _.isEqual
  );

  return ReactDOM.createPortal(
    <aside className="info-bar">
      {state.endTime && <Timer endTime={state.endTime} />}
      {state.activeVoting && <VoteDisplay usedVotes={state.usedVotes} possibleVotes={state.possibleVotes} />}
    </aside>,
    document.getElementById("root")!
  );
};
