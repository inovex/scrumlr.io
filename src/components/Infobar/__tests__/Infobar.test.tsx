import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import {InfoBar} from "../Infobar";
import {ApplicationState} from "types";
import getTestStore from "utils/test/getTestStore";
import {I18nextProvider} from "react-i18next";
import i18nTest from "i18nTest";
import {Router} from "react-router";
import {createBrowserHistory} from "history";

const renderInfoBar = (overwrite?: Partial<ApplicationState>) => {
  return render(
    <Router navigator={createBrowserHistory()} location={location}>
      <I18nextProvider i18n={i18nTest}>
        <Provider store={getTestStore(overwrite)}>
          <InfoBar />
        </Provider>
      </I18nextProvider>
    </Router>,
    {container: document.getElementById("root")!}
  );
};

describe("InfoBar", () => {
  beforeEach(() => {
    const root = global.document.createElement("div");
    root.setAttribute("id", "root");
    global.document.body.appendChild(root);
  });

  test("should have timer component during active timer", () => {
    const {container} = renderInfoBar({
      board: {
        status: "ready",
        data: {
          id: "test-board-id",
          name: "test-board-name",
          accessPolicy: "PUBLIC",
          showAuthors: true,
          showNotesOfOtherUsers: true,
          allowStacking: true,
          timerStart: new Date(123436789),
          timerEnd: new Date(123456789),
        },
      },
    });
    expect(container.getElementsByClassName("timer").length).toBe(1);
  });

  test("should have vote display component during active voting", () => {
    const {container} = renderInfoBar();
    expect(container.getElementsByClassName("vote-display").length).toBe(1);
  });

  test("should have return to focused not button during active sharing", () => {
    const {container} = renderInfoBar({
      board: {
        status: "ready",
        data: {
          id: "test-board-id",
          name: "test-board-name",
          accessPolicy: "PUBLIC",
          showAuthors: true,
          showNotesOfOtherUsers: true,
          allowStacking: true,
          sharedNote: "test",
        },
      },
    });
    expect(container.getElementsByClassName("info-bar__return-to-presented-note-button").length).toBe(1);
  });
});
