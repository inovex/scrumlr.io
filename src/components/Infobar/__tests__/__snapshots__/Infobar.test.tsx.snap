// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBar should render correctly with active voting and disabled timer 1`] = `
<div
  id="root"
>
  <aside
    class="info-bar"
  >
    <div
      class="vote-display"
    >
      <span
        title="VoteDisplay.tooltip"
      >
        0
         / 
        5
      </span>
    </div>
  </aside>
</div>
`;

exports[`InfoBar should render correctly with active voting and timer 1`] = `
<div
  id="root"
>
  <aside
    class="info-bar"
  >
    <div
      class="timer timer--expired"
      id="timer"
    >
      <span>
        00
        :
        00
      </span>
    </div>
    <div
      class="vote-display"
    >
      <span
        title="VoteDisplay.tooltip"
      >
        0
         / 
        5
      </span>
    </div>
  </aside>
</div>
`;

exports[`InfoBar should render correctly with disabled voting and active timer 1`] = `
<div
  id="root"
>
  <aside
    class="info-bar"
  >
    <div
      class="timer timer--expired"
      id="timer"
    >
      <span>
        00
        :
        00
      </span>
    </div>
  </aside>
</div>
`;

exports[`InfoBar should render correctly with disabled voting and timer 1`] = `
<div
  id="root"
>
  <aside
    class="info-bar"
  />
</div>
`;
